public/app/features/provisioning/Job/JobStatus.tsx

```diff
@@ -1,37 +1,53 @@
 import { skipToken } from '@reduxjs/toolkit/query';
-import { useEffect } from 'react';
+import { useEffect, useMemo } from 'react';
 
 import { Alert, ControlledCollapse, LinkButton, Spinner, Stack, Text } from '@grafana/ui';
-import { useGetRepositoryQuery } from 'app/api/clients/provisioning';
+import {
+  Job,
+  useGetRepositoryJobsWithPathQuery,
+  useGetRepositoryQuery,
+  useListJobQuery,
+} from 'app/api/clients/provisioning';
 
 import ProgressBar from '../Shared/ProgressBar';
-import { useRepositoryAllJobs } from '../hooks/useRepositoryAllJobs';
 import { getRepoHref } from '../utils/git';
 
 import { JobSummary } from './JobSummary';
 
 export interface JobStatusProps {
-  name: string;
+  watch: Job;
   onStatusChange?: (success: boolean) => void;
   onRunningChange?: (isRunning: boolean) => void;
   onErrorChange?: (error: string | null) => void;
 }
 
-export function JobStatus({ name, onStatusChange, onRunningChange, onErrorChange }: JobStatusProps) {
-  const [jobs, activeQuery, historicQuery] = useRepositoryAllJobs({ jobName: name, watch: true });
-  const job = jobs?.[0];
-
-  useEffect(() => {
-    if (onRunningChange) {
-      onRunningChange(
-        activeQuery.isLoading ||
-          historicQuery.isLoading ||
-          !job ||
-          job.status?.state === 'working' ||
-          job.status?.state === 'pending'
-      );
+export function JobStatus({ watch, onStatusChange, onRunningChange, onErrorChange }: JobStatusProps) {
+  const activeQuery = useListJobQuery({
+    fieldSelector: `metadata.name=${watch.metadata?.name}`,
+    watch: true,
+  });
+
+  // ??? ideally only execute this after the active query is done
+  const finishedQuery = useGetRepositoryJobsWithPathQuery({
```

Comment:
@Clarity-89 any better ideas?  ideally we only run this once the job is no longer in the activeQuery results.

This works, but is pretty ugly

Reply:
I see we're using `fieldSelector`, but what if we watch all the jobs instead and then get the ones we need from there?

Reply:
Ok, I see now that finished query is not in the list of active jobs, how about this: 
```ts
  const activeQuery = useListJobQuery({
    fieldSelector: `metadata.name=${watch.metadata?.name}`,
    watch: true,
  });
  const activeJob = activeQuery?.data?.items?.[0];
  const finishedQuery = useGetRepositoryJobsWithPathQuery(
    activeJob
      ? skipToken
      : {
          name: watch.metadata?.labels?.['provisioning.grafana.app/repository']!,
          uid: watch.metadata?.uid!,
        }
  );

  const job = activeJob || finishedQuery.data;
```
It'll not do a retry, but I think that can be done in a `useEffect` separately.